test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/BulkOperation/BulkOperationTest.cs

```diff
@@ -0,0 +1,172 @@
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNet.OData.Routing.Conventions;
+using Microsoft.Test.E2E.AspNet.OData.Common.Execution;
+using Microsoft.Test.E2E.AspNet.OData.Common.Extensions;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Test.E2E.AspNet.OData.BulkOperation
+{
+    public class BulkOperationTest : WebHostTestBase
+    {
+        public BulkOperationTest(WebHostTestFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            var controllers = new[] { typeof(EmployeesController), typeof(CompanyController), typeof(MetadataController) };
+            configuration.AddControllers(controllers);
+
+            configuration.Routes.Clear();
+            configuration.Count().Filter().OrderBy().Expand().MaxTop(null).Select();
+            configuration.MapODataServiceRoute("convention", "convention", BulkOperationEdmModel.GetConventionModel(configuration));
+            configuration.MapODataServiceRoute("explicit", "explicit", BulkOperationEdmModel.GetExplicitModel(configuration), new DefaultODataPathHandler(), ODataRoutingConventions.CreateDefault());
+            configuration.EnsureInitialized();
+        }
+
+        #region Post
+
+        [Fact]
+        public async Task PostEmployee_WithCreateFriends()
+        {
+            //Arrange
+
+            string requestUri = this.BaseAddress + "/convention/Employees";
+
+            var content = @"{
+                    'Name':'SqlUD',
+                    'Friends':[{ 'Id':1001, 'Name' : 'Friend 1001', 'Age': 31},{ 'Id':1002, 'Name' : 'Friend 1002', 'Age': 32},{ 'Id':1003, 'Name' : 'Friend 1003', 'Age': 33}]
+                     }";
+
+            var expectedContext = "{\"@odata.context\":\""+ this.BaseAddress + "/convention/$metadata#Employees/$entity\",\"ID\":0,\"Name\":\"SqlUD\",\"SkillSet\":[],\"Gender\":\"0\",\"AccessLevel\":\"0\",\"FavoriteSports\":null}";
+
+            var requestForPatch = new HttpRequestMessage(new HttpMethod("POST"), requestUri);
+
+            StringContent stringContent = new StringContent(content: content, encoding: Encoding.UTF8, mediaType: "application/json");
+            requestForPatch.Content = stringContent;
+
+            //Act & Assert
+            using (HttpResponseMessage response = await this.Client.SendAsync(requestForPatch))
+            {
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                var json = response.Content.ReadAsStringAsync().Result;
+                Assert.Contains("SqlUD", json);
+                Assert.Equal(expectedContext.ToString().ToLower(), json.ToString().ToLower());
+            }
+        }
+
+        [Fact]
+        public async Task PostEmployee_WithCreateFriendsFullMetadata()
```

Comment:
I think it would be better to have a more descriptive name, that expresses the intent of the test. Reading this, I don't know what Employee and Friends are and how they related. I have to scroll back to the model definition to find out that `Employee` is an entity type and `Friends` is a navigation property. And this test is trying to post an entity with related navigation properties.

Reply:
These tests were added with deep insert changes. I will get rid of them as we've removed changes for deep insert from this PR. so this will just be a normal POST with a normal POST response. Let me add more tests for patching./ deep update as that is what is being supported by this PR.